kb/2024-07-22/harvester_cluster_shutdown_and_restart.md

```diff
@@ -0,0 +1,311 @@
+authors:
+  - name: Jian Wang
+    title: Staff Software Engineer
+    url: https://github.com/w13915984028
+    image_url: https://github.com/w13915984028.png
+tags: [harvester, cluster, shutdown, rancher]
+hide_table_of_contents: false
+---
+
+Scenarios:
+
+1. The Harvester cluster is installed with 3+ nodes.
+
+1. The Rancher server is deoployed indepedently.
+
+1. The Harvester cluster is imported to Rancher server and worked as a node dirver
+
+1. Rancher deploys a couple of downstream k8s clusters, the machines/nodes of those clusters are backed by Harvester VMs.
+
+1. There are also some traditional VMs created on Harvester cluster, they have no direct connection with Rancher server.
+
+You have a plan to move those Harvester nodes geographically, it is essentail to shutdown the Harvester cluster and restart later.
+
+:::note
+
+2 3 4 are optional if your Harvester cluster is mainly running as IaaS. This instruction covers all the above scenarios
+
+:::
+
+## 1. Precondition
+
+### Generate a support-bundle file
+
+Follow [this instruction]https://docs.harvesterhci.io/v1.3/troubleshooting/harvester#generate-a-support-bundle to generate a support-bundle file.
+
+### Network Stability
+
+Harvester cluster is built on top of Kubernetes, a general requirement is that the Node/Host IP and the cluster VIP should keep stable in the whole lifecycle, if IP changes the cluster will fail to work as expected.
+
+If your VMs on Harvester are used as Rancher downstream cluster machines/nodes, and their IPs are allocated from DHCP server, also make sure those VMs will still get the same IPs after the Harvester cluster is rebooted and VMs are restarted.
+
+## 2. Backup
+
+### Backup all VMs if possible
+
+It is always a good practice to backup things before a whole cluster shutdown.
+
+### Backup downstream guest clusters if possible
+
+Harvester doesn't touch the downstream clusters' workload, you need to consider the related backup.
+
+## 3. Shutdown Workloads
+
+### 3.1 Shutdown Traditional VMs
+
+1. Shutdown VM from the VM shell (e.g. Linux `shutdown` command), the OS itself will save data to disks.
+
+2. Check the VM status from [Harvester UI - VM page](https://docs.harvesterhci.io/v1.3/troubleshooting/vm#vm-general-operations), when it is not `Off`, then click `Stop` command.
+
+### 3.2 Shutdown Rancher downstream cluster machines(VMs)
+
+Your Harvester cluster was [imported to Rancher](https://docs.harvesterhci.io/v1.3/rancher/virtualization-management) as a [node driver](https://docs.harvesterhci.io/v1.3/rancher/rancher-integration#creating-kubernetes-clusters-using-the-harvester-node-driver) before.
+
+When Rancher deploys a downstream cluster on node dirver Harvester, it creates a couple of VMs on Harvester automatically. Directly stopping those VMs on Harvester is not a good practice when Rancher is still managing the downstream cluster. For example, Rancher will create new VMs if you stop them from Harvester.
```

Comment:
we should check with rancher team about may be leveraging the patch capi cluster spec to pause the crd: https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/cluster_types.go#L49

one rancher may be managing multiple harvester clusters with downstream clusters spread across them. `pausing` reconcile of impacted clusters may be a better option than shutting down entire rancher cluster.